pandas/core/frame.py

```diff
@@ -850,7 +860,7 @@ def style(self):
         """
 
     @Appender(_shared_docs["items"])
-    def items(self):
+    def items(self) -> Generator[Tuple[str, Dtype], None, None]:
```

Comment:
A basic generator that only yields values can be annotated as having a return type of either Iterator[YieldType] or Iterable[YieldType]. https://mypy.readthedocs.io/en/latest/kinds_of_types.html#generators

my preference would be Iterable.